editors/code/package.json

```diff
@@ -1788,6 +1788,12 @@
                     "group": "navigation@1000"
                 }
             ]
-        }
+        },
+        "jsonValidation": [
+            {
+                "fileMatch": "rust-project.json",
+                "url": "https://json.schemastore.org/rust-project.json"
```

Comment:
From what I see in the VS Code repo, this setting will just say that `rust-project.json` should use the schema defined at that URL (the association happens [here](https://github.com/microsoft/vscode/blob/57f489b487d9434230321e37c32f3672df85acde/extensions/json-language-features/client/src/jsonClient.ts#L431)). However, downloading the schema does respect the user setting and is only done when [requesting the JSON content](https://github.com/microsoft/vscode/blob/57f489b487d9434230321e37c32f3672df85acde/extensions/json-language-features/client/src/jsonClient.ts#L274).

Reply:
And so to answer your question: No, this wouldn't work if the "JSON schema download" is disabled. This could be addressed by having the schema in the VS Code extension instead of in the schema store, but then other editors wouldn't be able to make use of the schema validation :(

Reply:
Or actually, yeah Rust-Analyzer could self-host the schema and make the one in the schema store point to it. We could do that if you want.

Reply:
Having it not work when downloading of json schemas is disabled is fine by me I guess. It is not a regression compared fo the current state and also not a privacy hole as I was worried about.

> Or actually, yeah Rust-Analyzer could self-host the schema and make the one in the schema store point to it. We could do that if you want.

I don't know how that would work, but if that is easy to do and doesn't have any downsides for either rust-analyzer or the schema store, it did be nice. I am not talking for the rust analyzer team, but personally I am fine with keeping it as is if there are downsides.

Reply:
@bjorn3 I don't think there would be any downsides to self-hosting the schema. It seems like it would be as simple as moving the schema to this repo and updating the link in the store registry: https://github.com/SchemaStore/schemastore/blob/3ffdbb43ddd21c1041caf85af3d646bbf2266abb/CONTRIBUTING.md#self-hosting-schemas. The schemas for `rust-toolchain` and `rustfmt` are in the store btw.

I'm fine either way. I'm not too sure of how it would work with other editor extensions that don't have the schema locally (I guess that's part of the schema store black magic). Would the request go to the store and then that would redirect to the GitHub URL registered there...?

Reply:
@bjorn3 Are you okay with me resolving this thread? Are we okay with keeping the schema in the store? :)

Reply:
I'm personally fine with keeping it in the store. @Veykril WDYT?

Reply:
No worries, thanks @Veykril for taking a look (and @bors for the discussion) <3